### ISSUE TITLE
Sadly only US

### ISSUE THREAD

**Issue by pbaerthel** [jjjonesjr33/petlibro#38]
### Device model name

Petlibro Dockstream

### Device model number

PLWF105

### Device type

Fountain

### Feature name

Europe or Germany Support

### Describe the feature

Sadly ist only Isabel with US Accounts. But I’m a German with German Petlibro account 

### What problem does this feature solve?

Add Germany server 

### Available to help with the feature development

- [X] I agree to help with testing or providing relevant information for this feature request

**Comment by jjjonesjr33** [comment]
@pbaerthel at the moment we only have the information for the US region. If you are able to provide the information for your regions API for us to test/add it that would help get the ball rolling on region support.

Please let me know if you are able to assist with this.

P.S. As of right now the workaround is to create a us account and migrate your devices to it.

**Comment by peterjeschke** [comment]
FIY I'm using a polar feeder from Germany and it's using the US API even with a German account

**Comment by jjjonesjr33** [comment]
@peterjeschke with your account, how was that setup originally? When I go to `https://de.petlibro.com/` it says invalid when I try to sign in. Are you able to sign into both the US site `https://petlibro.com/` and the Germany site `https://de.petlibro.com/`?

**Comment by jjjonesjr33** [comment]
@RolfKoenders, I’m curious if logging into the API works universally for any account type, regardless of the website or region it was created on. If that’s the case, it raises the question of why they don’t block creating duplicate accounts with the same information across their various websites.

From my testing, I can create accounts on any of the regional domains for Petlibro and update the region/location in the account settings. However, sharing devices across different regions/locations doesn’t seem to work.

It seems like further investigation may be necessary to determine exactly what works and where.

**Comment by peterjeschke** [comment]
I created my account using the app and selected "Germany" as my country of origin. The same credentials don't work for the website, neither petlibro.com nor de.petlibro.com.

**Comment by peterjeschke** [comment]
I was able to sign up at de.petlibro.com with the same email I used for the app _again_. I guess the shop/website accounts are completely separate from the app accounts.

**Comment by Denox78** [comment]
Hi everyone,

I'm from France and just ordered my first product

Should I create an account and register my product in the US region?

**Comment by peterjeschke** [comment]
> Hi everyone,
> 
> I'm from France and just ordered my first product
> 
> Should I create an account and register my product in the US region?

I created my account in the app and chose Germany, you should be fine choosing France

**Comment by Denox78** [comment]
> > Hi everyone,
> > I'm from France and just ordered my first product
> > Should I create an account and register my product in the US region?
> 
> I created my account in the app and chose Germany, you should be fine choosing France

Ok that's great, so you can see all your devices once added on the German's account ?

I'm also wondering if it's possible to delete all devices from an account to activate them on another (just in case I noticed everything's not working as expected)

**Comment by jjjonesjr33** [comment]
From what I’ve noticed, creating an account on the app is completely separate from their web domains. It looks like no matter which region you pick, as long as you create the account through the app, it lets you access the API.

That said, it might be worth testing a bit more to see if accounts from any region on the app work the same way.

**Comment by peterjeschke** [comment]
> > > Hi everyone,
> > > I'm from France and just ordered my first product
> > > Should I create an account and register my product in the US region?
> > 
> > I created my account in the app and chose Germany, you should be fine choosing France
> 
> Ok that's great, so you can see all your devices once added on the German's account ?

Yes, all devices are visible and shared between three accounts (all in the same region)

> I'm also wondering if it's possible to delete all devices from an account to activate them on another (just in case I noticed everything's not working as expected)

Haven't tried that

**Comment by peterjeschke** [comment]
I did some basic reverse engineering to search for host names in the Polar Feeder firmware and app to figure out if there are different endpoints for non-US scenarios. I couldn't find any in the firmware (either I didn't do it right or it's not stored in the firmware blob - maybe it's synced during the setup with the app).

But here are the (interesting) hostnames I could find:

```
mqtt.dl-aiot.com
m.designlibro.com.cn
api.designlibro.com.cn
api.us.petlibro.com
mqtt.us.petlibro.com
us.petlibro.com
```
dl-aiot.com redirects to petlibro.com.
I also found what looks to be fragments of host names:
```
-svc.dl-aiot.com
-api.dl-aiot.com
-m.dl-aiot.com
```
But domain names can't start with a hyphen, so something's missing. With some basic subdomain enumeration, I could verify that these subdomains _used to_ exist, but aren't available anymore:

```
test.m.dl-aiot.com
sit.m.dl-aiot.com
test.api.dl-aiot.com
```

Maybe the hyphen is just in there by mistake.

Looking in my DNS logs for dl-aiot.com, I can also find these host names:

```
us-mqtt-0.dl-aiot.com
oss.dl-aiot.com
```

All of this suggests that there is only one region (US) for the app.

**Comment by mavrster** [comment]
My original account is in New Zealand region and I get the login error

> File "/config/custom_components/petlibro/api.py", line 107, in request
>     raise PetLibroAPIError(f"Code: {data.get('code')}, Message: {data.get('msg')}")
> custom_components.petlibro.exceptions.PetLibroAPIError: Code: 1101, Message: MEMBER_NOT_EXIST

I created a new account in US region and got the same error.
